APITaxi/api/taxi.py

```diff
@@ -114,106 +113,52 @@ def put(self, taxi_id):
+        This function gets the taxi's zones of the position sent in the request.
+        And then gets taxis around this position and filter them with the following rules:
+         - Is the taxi available?
+         - Is the position fresh enough?
+         - Can this taxi operate here?
+        Then it will gets some information from the database.
+        The taxis will be ordered by growing distance from the request's position
+        """
         p = get_parser.parse_args()
+        t = time()
         lon, lat = p['lon'], p['lat']
-        if current_app.config['LIMITED_ZONE'] and\
-            not Point(lon, lat).intersects(current_app.config['LIMITED_ZONE']):
-            #It must be 403, but I don't know how our clients will react:
-            return {'data': []}
-        self.zupc_customer = db.session.execute(
-            """SELECT id, parent_id, max_distance, insee
-               FROM "ZUPC"
-               WHERE ST_INTERSECTS(shape, 'POINT(:lon :lat)')
-               AND parent_id = id
-               ORDER BY max_distance ASC;""", {"lon": lon, "lat": lat}
-        ).fetchall()
-        if len(self.zupc_customer) == 0:
-            current_app.logger.debug('No zone found at {}, {}'.format(lat, lon))
+        zupc_customer = models.ZUPC.get(lon, lat)
+        if not zupc_customer:
             return {'data': []}
-        zupc_insee = self.zupc_customer[0][3]
-#We can deactivate the max radius for a certain zone
-        inactive_filter_period = current_app.config['INACTIVE_FILTER_PERIOD']
-        hour = datetime.now().hour
-        if inactive_filter_period[0] > inactive_filter_period[1]:
-            is_inactive = hour >= inactive_filter_period[0] or\
-                    hour <= inactive_filter_period[1]
-        else:
-            is_inactive = inactive_filter_period[0] <= hour <= inactive_filter_period[1]
-        if is_inactive:
-            max_distance = current_app.config['DEFAULT_MAX_RADIUS']
-        else:
-            max_distance = min([v for v in [v[2] for v in self.zupc_customer] if v and v>0]
-                           + [current_app.config['DEFAULT_MAX_RADIUS']])
-        self.check_freshness()
-        g.keys_to_delete = []
-        name_redis = '{}:{}:{}'.format(lon, lat, time())
-        g.keys_to_delete.append(name_redis)
-        #It returns a list of all taxis near the given point
-        #For each taxi you have a tuple with: (id, distance, [lat, lon])
-        nb_positions = redis_store.georadius(current_app.config['REDIS_GEOINDEX_ID'],
-                lon, lat, radius=max_distance, unit='m', store_dist=name_redis)
-        if nb_positions == 0:
-            current_app.logger.debug('No taxi found at {}, {}'.format(lat, lon))
+        max_distance = models.ZUPC.get_max_distance(zupc_customer)
+        self.clean_taxis_timestamps()
+        positions_redis = '{}:{}:{}'.format(lon, lat, t)
+        if models.TaxiRedis.store_positions(lon, lat, max_distance, t, redis_store, positions_redis) == 0:
```

Comment:
`models.TaxiRedis.store_positions` returns 0 if `nb_positions == 0`, but nb_positions is always a list (which can be empty).

This equality nerver returns true.

The pythonic way to compare to an empty value is to check `if not xxx` or `if xxx` instead of `if xxx == ...`

Note: APITaxi_models must also be updated.